firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -400,10 +400,12 @@ public void testUpdateTransactionally() {
 
     // Let all of the transactions fetch the old value and stop once.
     waitFor(Tasks.whenAll(readTasks));
-    assertEquals(3, started.intValue());
-    // Let all of the transactions continue and wait for them to finish.
+    // There should be 3 initial transaction runs.
+    assertEquals(3, counter.get());
     barrier.setResult(null);
     waitFor(Tasks.whenAll(transactionTasks));
+    // There should be a total of 3 retries: once for the 2nd update, and twice for the 3rd update.
```

Comment:
Why are we guaranteed that the 3rd one will retry twice?  This seems timing-dependent.  If the 2nd update happens really fast, the 3rd update might only retry once.

In practice, maybe this is relatively reliable, but I worry it could result in a flakey test...  At bare minimum I would add a comment that this isn't a guarantee and we should revisit if this test shows up as flakey.

Or am I missing something?

Reply:
That comment was mostly based on me running the integration test and seeing that the the 3rd transaction consistently retried twice, but duly noted, will update the comment accordingly.